bench/benches/operators.rs

```diff
@@ -0,0 +1,178 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use nb2::batch::Either;
+use nb2::packets::{Ethernet, Packet};
+use nb2::testils::criterion::BencherExt;
+use nb2::testils::proptest::*;
+use nb2::{compose, Batch, Mbuf};
+use proptest::prelude::*;
+use proptest::strategy;
+
+const BATCH_SIZE: usize = 500;
+static mut COUNTER: u32 = 0;
+
+fn filter_true(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Mbuf> {
+    batch.filter(|_p| true)
+}
+
+fn filter_false(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Mbuf> {
+    batch.filter(|_p| false)
+}
+
+fn filter_map(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Ethernet> {
+    batch.filter_map(|p| {
+        let ethernet = p.parse::<Ethernet>()?;
+        Ok(Either::Keep(ethernet))
+    })
+}
+
+fn map_then_filter(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Ethernet> {
+    batch.map(|p| p.parse::<Ethernet>()).filter(|_p| true)
+}
+
+fn map(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Ethernet> {
+    batch.map(|p| p.parse::<Ethernet>())
+}
+
+fn group_by(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Mbuf> {
+    unsafe { COUNTER += 1 };
```

Comment:
but don't we want the counter to go up on each call to routine(batch), instead of be re-instantiated?

Reply:
it won't be reinitialized if it's static. it will be [once](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7ed7350a3a1d725a126d01e64d0b2d23) for that fn.